mobile/ios/ios-core.ts

```diff
@@ -1084,33 +1090,71 @@ export class GDBServer implements Mobile.IGDBServer {
 			}
 		}
 	}
-
-	public run(argv: string[]): void {
-		this.send("QStartNoAckMode");
-		this.socket.write("+");
-		this.send("QEnvironmentHexEncoded:");
-		this.send("QSetDisableASLR:1");
-
-		let encodedArguments = _.map(argv, (arg, index) => util.format("%d,%d,%s", arg.length * 2, index, new Buffer(arg).toString("hex"))).join(",");
-		this.send("A" + encodedArguments);
-
-		this.send("qLaunchSuccess");
-
-		if(this.$hostInfo.isWindows) {
-			this.send("vCont;c");
-		} else {
-			if (!this.$options.justlaunch) {
-				this.socket.pipe(new GDBStandardOutputAdapter()).pipe(process.stdout);
-				this.socket.pipe(new GDBSignalWatcher());
+	
+	public init(): IFuture<void> {
+		return (() => {
+			this.send("QStartNoAckMode");
+			this.socket.write("+");
+			this.send("QEnvironmentHexEncoded:");
+			this.send("QSetDisableASLR:1");
+		}).future<void>()();
+	}
+	
+	public run(argv: string[]): IFuture<void> {
+		return (() => {
+			this.init().wait();
+			
+			let encodedArguments = _.map(argv, (arg, index) => util.format("%d,%d,%s", arg.length * 2, index, new Buffer(arg).toString("hex"))).join(",");
```

Comment:
I see new `Buffer(...).toString("hex")` repeated a couple of times. Consider extracting it into a method.